flamingpy/utils/viz.py

```diff
@@ -582,16 +561,89 @@ def _get_node_color(egraph, color_nodes, point):
                 "the dictionary keys to valid attribute values and"
                 "dictionary values to valid matplotlib color strings."
             )
-        node_property = egraph.nodes[point].get(node_attribute)
+        node_property = egraph.nodes[node].get(node_attribute)
         color = color_dict.get(node_property)
 
     elif color_nodes == True:
-        color = egraph.nodes[point].get("color") or "black"
+        color = egraph.nodes[node].get("color") or "black"
     else:
         color = "black"
     return color
 
 
+def _get_node_info(egraph, node, information="coordinates"):
+    """Information to be displayed when hovering over a node based on
+    ``color_nodes``
+
+    Arguments:
+        egraph (EGraph): the EGraph with the node of interest.
+        node (tuple): the node to get the information from.
+        information (str, iterable or NoneType): the information to be displayed:
+            - if ``information`` is a string, the value of the node attribute ``information``,
+            - if ``information`` is an iterable, a list of the values of the node attributes in ``information``,
+            - if ``information`` is None, the coordinates of the node.
+    """
+    if information == "coordinates":
+        return str(node)
```

Comment:
Quick request: let's add an index plotting option. Would this work?

```suggestion
    if information == "coordinates":
        return str(node)
    elif information == "index":
        return str(self.to_indices[node])
```

Reply:
I included it as an option!

Reply:
I might still need to add some tests for it